### PR TITLE
61455 add aux component info

### PR DESCRIPTION
**Pull request by nilsgstrabo**
Add oauth2 replica info to component model
Get logs for an aux replica
Restart aux service

### CODE REVIEW

File: api/environments/component_handler.go [hunk 1]
```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
 	return nil
 }
 
+// RestartComponentAuxiliaryResource Restarts a component's auxiliary resource
+func (eh EnvironmentHandler) RestartComponentAuxiliaryResource(appName, envName, componentName, auxType string) error {
+	log.Infof("Restarting auxiliary resource %s for component %s, %s", auxType, componentName, appName)
+
+	deploySummary, err := eh.deployHandler.GetLatestDeploymentForApplicationEnvironment(appName, envName)
```
Comment (satr) [inline api/environments/component_handler.go:138]: This call created deployment with components and jobs (using `deploy.getDeployments(namespace, appName, "", latest)`, with `latest: true`), while just only last (active) deployment name is needed, to repeat the loop in the next operator `eh.deployHandler.GetDeploymentWithName(appName, deploySummary.Name)`, which also use `deploy.getDeployments(namespace, appName, "", latest)`, but with `latest: false`. 
Maybe we can reduce this complexity?

File: api/environments/component_handler.go [hunk 2]
```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
...
+
+	if auxResourceDto.Deployment == nil || auxResourceDto.Deployment.Status != deploymentModels.ConsistentComponent.String() {
+		currentStatus := deploymentModels.ComponentReconciling.String()
+		if auxResourceDto.Deployment != nil {
+			currentStatus = auxResourceDto.Deployment.Status
+		}
+		return environmentModels.CannotRestartComponent(appName, componentName, currentStatus)
+	}
+
+	selector := labelselector.ForAuxiliaryResource(appName, componentName, auxType).String()
```
Comment (satr) [inline api/environments/component_handler.go:166]: Can the logic just start from here? With error `environmentModels.NonExistingComponent(appName, componentName)` if the deployment does not exist?
I tried to watch deployment status `k get deploy  -n app-namespace www -o json|jq -r .status`
while running commands
`k delete pods --all -n app-namespace`
`k rollout restart deployment www  -n app-namespace`
Status was changed , but a bit unclear way. So I'm not sure if it is good idea.

File: api/environments/component_handler.go [hunk 3]
```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
...
+	return eh.patchDeploymentForRestart(&deployments.Items[0])
+}
+
+func (eh EnvironmentHandler) patchDeploymentForRestart(deployment *appsv1.Deployment) error {
+	if deployment.Spec.Template.Annotations == nil {
+		deployment.Spec.Template.Annotations = make(map[string]string)
+	}
+
+	deployment.Spec.Template.Annotations[restartedAtAnnotation] = radixutils.FormatTimestamp(time.Now())
+	if _, err := eh.client.AppsV1().Deployments(deployment.Namespace).Update(context.TODO(), deployment, metav1.UpdateOptions{}); err != nil {
```
Comment (satr) [inline api/environments/component_handler.go:185]: Should we use the method like in the `radix-operator` to update status:
`retry.RetryOnConflict(retry.DefaultRetry, func() error {...}`

File: api/environments/environment_controller.go [hunk 4]
```diff
@@ -862,6 +872,65 @@ func RestartApplication(accounts models.Accounts, w http.ResponseWriter, r *http
 	radixhttp.JSONResponse(w, r, "Success")
 }
 
+// RestartComponent Restarts job
+func RestartComponentAuxiliaryResource(accounts models.Accounts, w http.ResponseWriter, r *http.Request) {
+	// swagger:operation POST /applications/{appName}/environments/{envName}/components/{componentName}/aux/{auxType}/restart component restartComponentAuxiliaryResopurce
```
Comment (satr) [inline api/environments/environment_controller.go:877]: Typo in `restartComponentAuxiliaryResopurce`

File: api/pods/pod_handler.go [hunk 5]
```diff
@@ -43,12 +45,28 @@ func (ph PodHandler) HandleGetEnvironmentScheduledJobLog(appName, envName, sched
 	return ph.getScheduledJobLog(envNs, scheduledJobName, containerName, sinceTime)
 }
 
+// HandleGetEnvironmentPodLog Get logs from pod in environment
```
Comment (satr) [inline api/pods/pod_handler.go:48]: Method name in comment does not match to real name `HandleGetEnvironmentAuxiliaryResourcePodLog`

**Review by satr — CHANGES_REQUESTED** [review]

File: api/deployments/models/component_deployment.go [hunk 6]
```diff
@@ -90,9 +90,57 @@ type Component struct {
...
+		}
+	}
+
+	return nil
+}
+
+// AuxiliaryResource describes an auxiliary resourses for a component
+// swagger:model AuxiliaryResource
+type AuxiliaryResource struct {
+	// Type of auxliliary resource
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply api/deployments/models/component_deployment.go:113]: fixed

File: api/deployments/models/component_deployment.go [hunk 7]
```diff
@@ -90,9 +90,57 @@ type Component struct {
...
+	for _, aux := range c.AuxiliaryResources {
+		if aux.Type == auxType {
+			return &aux
+		}
+	}
+
+	return nil
+}
+
+// AuxiliaryResource describes an auxiliary resourses for a component
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply api/deployments/models/component_deployment.go:110]: fixed

File: api/deployments/models/component_builder.go [hunk 8]
```diff
@@ -37,6 +38,7 @@ type componentBuilder struct {
 	ports                     []Port
 	schedulerPort             *int32
 	scheduledJobPayloadPath   string
+	auxResources              []AuxiliaryResource
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply api/deployments/models/component_builder.go:41]: Agree. Refactored structs.

File: api/environments/component_handler.go (hunk 1 above)
Comment (nilsgstrabo) [reply api/environments/component_handler.go:138]: Changed to use another method

File: api/environments/component_handler.go (hunk 3 above)
Comment (nilsgstrabo) [reply api/environments/component_handler.go:208]: Agree. Fixed.

File: api/environments/environment_controller.go (hunk 4 above)
Comment (nilsgstrabo) [reply api/environments/environment_controller.go:877]: fixed

File: api/environments/environment_controller.go [hunk 9]
```diff
@@ -1015,3 +1084,87 @@ func GetScheduledJobLog(accounts models.Accounts, w http.ResponseWriter, r *http
...
+	//   in: header
+	//   description: Works only with custom setup of cluster. Allow impersonation of test group (Required if Impersonate-User is set)
+	//   type: string
+	//   required: false
+	// responses:
+	//   "200":
+	//     description: "pod log"
+	//     schema:
+	//        type: "string"
+	//   "404":
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply api/environments/environment_controller.go:1140]: Fixed, except 409. We're not updating anything, so no conflic should occur

File: api/pods/pod_handler.go (hunk 5 above)
Comment (nilsgstrabo) [reply api/pods/pod_handler.go:48]: fixed

**Review by satr — APPROVED** [review]

File: api/environments/component_handler.go [hunk 10]
```diff
@@ -126,6 +132,82 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
...
+
+	return eh.patchDeploymentForRestart(&deployments.Items[0])
+}
+
+func canDeploymentBeRestarted(deployment *appsv1.Deployment) bool {
+	if deployment == nil {
+		return false
+	}
+
+	if deploymentModels.ComponentStatusFromDeployment(deployment) != deploymentModels.ConsistentComponent {
```
[the remark this answers is not in the document]
Comment (nilsgstrabo) [reply api/environments/component_handler.go]: fixed

**Pull request merged**